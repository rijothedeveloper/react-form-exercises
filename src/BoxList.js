import { useState } from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";

const BoxList = () => {
    const getInitialBoxState = () => {
        const boxes = [{background: 'blue', width: 100, height: 100}]
        return boxes;
    }
    const [boxes, setBoxes] = useState(getInitialBoxState)
    const addBox = (boxObj) => setBoxes([...boxes, boxObj])
    const boxArray = boxes.map(e => <Box backgroundColor={e.background} width={e.width} height= {e.height} />)
    return (
        <div>
            <div>      
            {boxArray}
            </div>
           
            <NewBoxForm addBox={addBox} />
        </div>
    )
}

export default BoxList;