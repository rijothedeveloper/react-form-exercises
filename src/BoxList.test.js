import React from "react";
import {render} from "@testing-library/react";
import BoxList from "./BoxList"

// smoke test
it("renders without crashing", function() {
    render(<BoxList />)
})

// snapshot test
it("matches snapshot", () => {
    const {asFragment} = render(<BoxList />)
    expect(asFragment()).toMatchSnapshot();
});