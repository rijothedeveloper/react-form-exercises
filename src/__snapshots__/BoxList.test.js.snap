// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        data-id="2ff757ea-ad2e-42ca-8096-a64a05be9b99"
        style="background-color: blue; height: 100px; width: 100px;"
      >
        <button>
          close
        </button>
      </div>
    </div>
    <form>
      <label
        for="width"
      >
        Width: 
      </label>
      <input
        id="width"
        name="width"
        type="text"
        value=""
      />
      <label
        for="height"
      >
        Height: 
      </label>
      <input
        id="height"
        name="height"
        type="text"
        value=""
      />
      <label
        for="background"
      >
        Background color: 
      </label>
      <input
        id="background"
        name="background"
        type="text"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
